### ISSUE TITLE
Existing Magento cart not loading

### ISSUE THREAD

**Issue by jimcreate78** [DivanteLtd/vue-storefront#1762]
Upon logging in with a user I'm no longer seeing the cart restore or the application-after-loaded event fire.  

Previously, when a user would login the cart would be restored and the event would fire.

This started after updating to v1.3.1 from v1.1.0

local.json:
` "cart": {
      "multisiteCommonCart": false,
      "server_merge_by_default": false,
      "synchronize": true,
      "synchronize_totals": true,
}`

**Comment by pkarw** [comment]
Please try to comment out the the condition of https://github.com/pkarw/vue-storefront/blob/1e4239cbdf5b727ee042d6bff740fdfcdc11d9d6/core/store/modules/cart/actions.ts#L470

In other words - just keep the 
```js
        rootStore.dispatch('cart/serverCreate', { guestCart: false }, { root: true })
```

in this `userAfterLoggedin` - and let me know if it helped. If so - I'll add a fix to that

**Comment by jimcreate78** [comment]
@pkarw servercart-after-diff fires now but it didn't restore the Magento cart.  The cart state is empty.

**servercart-after-diff payload:**
`{"diffLog":[{"party":"client","sku":"WB-CLM-1","status":"no_item"},{"party":"client","sku":"CC-CCP-05","status":"no_item"}],"serverItems":[{"item_id":1052,"sku":"WB-CLM-1","qty":3,"name":"Clem","price":14.5,"product_type":"configurable","quote_id":"26","product_option":{"extension_attributes":{"configurable_item_options":[{"option_id":"177","option_value":13}]}}},{"item_id":1060,"sku":"CC-CCP-05","qty":3,"name":"Premium","price":45,"product_type":"configurable","quote_id":"26","product_option":{"extension_attributes":{"configurable_item_options":[{"option_id":"177","option_value":36}]}}}],"clientItems":[],"dryRun":true,"event":{"url":"http://ec2-54-219-238-247.us-west-1.compute.amazonaws.com:8080/api/cart/pull?token={{token}}&cartId={{cartId}}","payload":{"method":"GET","headers":{"Content-Type":"application/json"},"mode":"cors"},"silent":true,"force_client_state":false,"dry_run":true,"callback_event":"store:cart/servercartAfterPulled","task_id":"1537914696178-05bcecb8-e3e7-effb-b7c3-0c2b2a914c4f","transmited":true,"created_at":"[native Date Tue Sep 25 2018 15:31:36 GMT-0700 (Pacific Daylight Time)]","updated_at":"[native Date Tue Sep 25 2018 15:31:36 GMT-0700 (Pacific Daylight Time)]","transmited_at":"[native Date Tue Sep 25 2018 15:31:37 GMT-0700 (Pacific Daylight Time)]","result":[{"item_id":1052,"sku":"WB-CLM-1","qty":3,"name":"Clem","price":14.5,"product_type":"configurable","quote_id":"26","product_option":{"extension_attributes":{"configurable_item_options":[{"option_id":"177","option_value":13}]}}},{"item_id":1060,"sku":"CC-CCP-05","qty":3,"name":"Premium","price":45,"product_type":"configurable","quote_id":"26","product_option":{"extension_attributes":{"configurable_item_options":[{"option_id":"177","option_value":36}]}}}],"resultCode":200,"code":200,"acknowledged":false}}`

**Comment by pkarw** [comment]
This is correct behaviour. If You've got set `config.cart.server_merge_by_default=true` then there is just this event (`servercart-after-diff`) executed and You supposed to display the popup or whatever to let the users decide if they like to merge cart or not (and merge it manually) :-)

**Comment by pkarw** [comment]
If You'll set `config.cart.server_merge_by_default=false` in config then automerge will be executed without firing `servercart-after-diff`

**Comment by jimcreate78** [comment]
The server merge by default was functionality I requested because we didn't want to keep merging carts together.  In 1.1.0 the cart restored from Magento and we did our work on servercart-after-pulled.  Now the cart doesn't restore from Magento and that event doesn't fire.

**Comment by pkarw** [comment]
Hmm afaik nothing has been changed regarding this. Please let me know which value has you got in local.json: config.cart.server_merge_by_default

**Comment by pkarw** [comment]
Plus please do confirm that You have this fix: 
https://github.com/DivanteLtd/vue-storefront/commit/f9a9ddcafd5ca9c3ad3d46fc1e8217dd3592b74 merged in?3

**Comment by jimcreate78** [comment]
> local.json:
"cart": { "multisiteCommonCart": false, "server_merge_by_default": false, "synchronize": true, "synchronize_totals": true, }

I have the latest development branch currently with f9a9ddc included.

**Comment by pkarw** [comment]
Ok, I’m not sure if I get it right. Because auto merge will work just fine if Yoi set server_merge_by_default to true - and ... nothing has been changed since this option was introduced hmmmm

**Comment by pkarw** [comment]
https://github.com/DivanteLtd/vue-storefront/pull/1131#issuecomment-424792415 - with the latest comment + nothing has changed there since then

**Comment by jimcreate78** [comment]
We added the server_merge_by_default so that when the Magento cart loads it wouldn't merge in whats in the client cart.  This variable wasn't controlling if the cart is restored or not but if it's merged or not.

**Comment by pkarw** [comment]
In other words: after user logs in you would like to restore the server content by default (replacing current cart) and if there were any items in the cart that we should fire after-diff to give the chance to merge these items back in to th  cart. Correct?

This behavior hasn’t been implemented so far. If it worked this way like restoring server cart even when server_merge_vy_default = false it must have been by some side effect because it’s impossible:) when I wrote this feature my assumption was that restore = merge with empty cart so this operation never was executed automatically.

But nevermjnd, please let me know if this is what you’re looking for and I belive we can easily add this feature to the core with separate option: “server_restore_by_default” :-)

You can also try to Add sth like:

rootStore.dispatch(„cart/clear”)
rootStore.dispatch('cart/serverPull', { forceClientState: false, dryRun: false }, { root: true })

In servercart-after-diff and you’ll get pretty much the same effect
